### PR TITLE
Refactor documents

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,10 +29,14 @@ To add a dependency using Maven:
 ## Docs
 * [Getting started](docs/getting-started.md)
 * [Java API Guide](docs/api-guide.md)
-* [Scalar DB supported databases](docs/scalardb-supported-databases.md)
+* [Scalar DB Samples](https://github.com/scalar-labs/scalardb-samples)
```
Comment (brfrn169) [inline README.md:32]: Reorganized this document index.

File: docs/api-guide.md [hunk 2]
```diff
@@ -734,4 +734,4 @@ If you catch `ValidationConflictException`, like the `CrudConflictException` cas
 * [Getting started](getting-started.md)
 * [Multi-storage Transactions](multi-storage-transactions.md)
 * [Two-phase Commit Transactions](two-phase-commit-transactions.md)
-* [Scalar DB server](scalardb-server.md)
+* [Scalar DB Server](scalardb-server.md)
```
Comment (brfrn169) [inline docs/api-guide.md:737]: Replaced `Scalar DB server` with `Scalar DB Server` for consistency.

File: docs/scalardb-server.md [hunk 3]
```diff
@@ -143,4 +143,9 @@ scalar.db.grpc.deadline_duration_millis=60000
 
 Please see the following sample to learn Scalar DB Server further:
 
-[Scalar DB Server Sample](https://github.com/scalar-labs/scalardb-samples/tree/main/scalardb-server-sample)
+- [Scalar DB Server Sample](https://github.com/scalar-labs/scalardb-samples/tree/main/scalardb-server-sample)
+
+Please also see the following documents to learn how to deploy Scalar DB Server:
+
+- [Deploy Scalar DB Server on AWS](https://github.com/scalar-labs/scalar-kubernetes/blob/master/docs/ManualDeploymentGuideScalarDBServerOnEKS.md)
+- [Deploy Scalar DB Server on Azure](https://github.com/scalar-labs/scalar-kubernetes/blob/master/docs/ManualDeploymentGuideScalarDBServerOnAKS.md)
```
Comment (brfrn169) [inline docs/scalardb-server.md:151]: Added links to the `Deploy Scalar DB Server` documents.

**Review by Torch3333 — COMMENTED** [review]
Thank you, it looks good. 
I left some suggestions, please have a look.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by brfrn169** [comment]
@Torch3333 Thank you for pointing out my silly mistakes. I fixed them in the latest commit. Please take a look again when you have time!

**Review by Torch3333 — APPROVED** [review]
Thank you for the revision, LGTM!

**Pull request merged**